### PR TITLE
Ws/feature/subscription controller and route

### PR DESCRIPTION
**Pull request by WeiyuSun**
subscriber route and controller created

### CODE REVIEW

File: backend/route/user-route.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 router.get("/profile", userController.getUserProfile)
 router.post("/profile", userController.updateUserProfile)
 router.get("/delete_account", userController.removeAccount)
 router.post("/username", userController.updateUsername)
 router.post("/password", userController.updatePassword)
+router.post("/subscription/followNewUser", userMiddleware.subscriptionValidation, userController.subscribeUser)
+router.get("/subscription/getFollowing", userController.getMyFollowing)
+router.get("/subscription/getAudience", userController.getMyAudience)
+router.get("/subscription/setNotification", userController.setNotification)
+router.get("/subscription/cancel", userMiddleware.unsubscriptionValidation, userController.cancelSubscription)
```
Comment (orangeboy55555) [inline backend/route/user-route.js:14]: Can you create a route that will tell me if the user has subscribed to another user? That would be helpful. Thanks.

**Review by orangeboy55555 — CHANGES_REQUESTED** [review]

File: backend/route/user-route.js (hunk 1 above)
Comment (WeiyuSun) [reply backend/route/user-route.js:14]: yes

Comment (orangeboy55555) [reply backend/route/user-route.js:14]: Thanks a bunch

**Review by orangeboy55555 — APPROVED** [review]

**Review by DaTanUmanitoba — APPROVED** [review]
It seems all good to me.

**Pull request merged**